### PR TITLE
[FIX] analytic: group by with domain including `analytic_distribution`

### PR DESCRIPTION
**Pull request by abla001**
To reproduce
============
on journal items add filter (Analytic contains something) and group by Account, no lines will be shown.

Problem
=======
`analytic_distribution` is a Json field stored as string in database as follows: `{"id" : "percentage"}`, which explains why no result is returned as there is no trace for Analytics name.

Solution
========
following same [fix](https://github.com/odoo/odoo/commit/bc5b72937a1c79688f7f6dc1c8aeaa7eb4cb6744), we replace `analytic_distribution` by `analytic_distribution_search` in the domain of `read_group`.

### CODE REVIEW

**Review by gawa-odoo — COMMENTED** [review]
Hello!
It seems fine! I would just put it just below the search function, as it follows the same logic, but that's just a preference thing

File: addons/analytic/models/analytic_mixin.py [hunk 1]
```diff
@@ -69,6 +69,13 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
                 arg[0] = 'analytic_distribution_search'
         return super()._search(args, offset, limit, order, count, access_rights_uid)
 
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
+        for arg in domain:
+            if isinstance(arg, (list, tuple)) and arg[0] == 'analytic_distribution' and isinstance(arg[2], str):
+                arg[0] = 'analytic_distribution_search'
```
[the remark this answers is not in the document]
Comment (william-andre) [reply addons/analytic/models/analytic_mixin.py:76]: Probably need to change it for `_search` too... (maybe worth to refactor using a helper function?)

**Comment by abla001** [comment]
@william-andre thank you for the feedback, I applied your suggestion, I couldn't find good name for the helper function :sweat_smile:  if you have better suggestion I take it

**Comment by abla001** [comment]
@william-andre can you take a look at the update please ?

**Comment by abla001** [comment]
@fw-bot r+

**Pull request closed without merging**